### ISSUE TITLE
azuredevops_build_definition import crash

### ISSUE THREAD

**Issue by dktodorov** [microsoft/terraform-provider-azuredevops#315]
### Terraform (and Azure DevOps Provider) Version
Terraform 0.14.7
azuredevops 0.1.2

### Affected Resource(s)

* `azuredevops_build_definition`

### Terraform Configuration Files

```hcl
terraform {
  backend "azurerm" {}
  required_providers {
    azuredevops = {
      source = "microsoft/azuredevops"
      version = ">=0.1.0"
    }
  }
}

provider "azuredevops" {
  org_service_url = "https://dev.azure.com/organization/"
}

data "azuredevops_project" "project" {
  name = "Project-Name"
}

data "azuredevops_git_repository" "repo" {
  project_id = data.azuredevops_project.project.id
  name       = "TestRepo"
}

resource "azuredevops_build_definition" "build" {
  project_id = data.azuredevops_project.project.id
  name       = "Sample Build Definition"
  path       = "\\Path"

  ci_trigger {
    use_yaml = true
  }

  repository {
    repo_type   = "TfsGit"
    repo_id     = data.azuredevops_git_repository.repo.id
    branch_name = "develop"
    yml_path    = "azure-pipelines.yml"
  }
}
```

### Debug Output
https://gist.github.com/dktodorov/52f130a418f2d47203ad380b120296a2#file-gistfile1-txt

### Panic Output
https://gist.github.com/dktodorov/52f130a418f2d47203ad380b120296a2#file-gistfile2-txt

### Expected Behavior
Import Resouce

### Actual Behavior
Terraform Crash

### Steps to Reproduce

1. `terraform import`

### Important Factoids
No

**Comment by xuzhang3** [comment]
Hi @dktodorov   I cannot reproduce your error, can you find the  `report build status` configuration in the pipeline you try to import?
![image](https://user-images.githubusercontent.com/57888764/109773234-f1676f80-7c39-11eb-9fb7-be4dffcbd1c9.png)

**Comment by dktodorov** [comment]
Hi @xuzhang3,

our pipelines were initially created with Python SDK https://github.com/microsoft/azure-devops-python-api/tree/dev/azure-devops/azure/devops/v6_0

When you create the pipeline with the SDK, **buildDefinition.Repository.Properties** can be **nil**. So properties dictionary didn't exist at all.
Maybe it is better to add a fix, which actually checks if properties is a dictionary, before trying to access its keys: https://github.com/microsoft/terraform-provider-azuredevops/blob/master/azuredevops/internal/service/build/resource_build_definition.go#L502

**Comment by xuzhang3** [comment]
Hi @dktodorov   The python SDK you use is v6, but ADO provider is based on v5.  I can add a check for `buildDefinition.Repository.Properties`, but this might not help in this scenario.

**Comment by xuzhang3** [comment]
Fixed in v0.1.3, closing this issue. Feel free to open another issue if you have questions.
